src/Compilers/CSharp/Portable/Lowering/LocalRewriter/LocalRewriter_Call.cs

```diff
@@ -364,6 +371,155 @@ private static bool IsSafeForReordering(BoundExpression expression, RefKind kind
             }
         }
 
+        private ImmutableArray<BoundExpression> VisitArguments(
+            SyntaxNode syntax,
+            ImmutableArray<BoundExpression> arguments,
+            Symbol methodOrIndexer,
+            ImmutableArray<int> argsToParamsOpt,
+            ImmutableArray<RefKind> argumentRefKindsOpt,
+            [NotNullIfNotNull("rewrittenReceiver")] ref BoundExpression? rewrittenReceiver,
+            out ArrayBuilder<LocalSymbol>? temps)
+        {
+            Debug.Assert(argumentRefKindsOpt.IsDefault || argumentRefKindsOpt.Length == arguments.Length);
+            var requiresInstanceReceiver = methodOrIndexer.RequiresInstanceReceiver() && methodOrIndexer is not MethodSymbol { MethodKind: MethodKind.Constructor } and not FunctionPointerMethodSymbol;
+            Debug.Assert(!requiresInstanceReceiver || rewrittenReceiver != null || _inExpressionLambda);
+            temps = null;
+            var positionsAssignedToTemp = BitVector.Null;
+
+            if (arguments.IsEmpty)
+            {
+                return arguments;
+            }
+
+            var visitedArgumentsBuilder = ArrayBuilder<BoundExpression>.GetInstance(arguments.Length);
+            var parameters = methodOrIndexer.GetParameters();
+            var receiverStored = false;
+
+            for (int i = 0; i < arguments.Length; i++)
+            {
+                var argument = arguments[i];
+                if (argument is BoundDiscardExpression discard)
+                {
+                    ensureTempTrackingSetup(ref temps, ref positionsAssignedToTemp);
+                    visitedArgumentsBuilder.Add(_factory.MakeTempForDiscard(discard, temps));
+                    positionsAssignedToTemp[i] = true;
+                    continue;
+                }
+
+                ImmutableArray<BoundInterpolatedStringArgumentPlaceholder> argumentPlaceholders = setupInterpolationDataIfNeeded(
+                    i,
+                    ref receiverStored,
+                    ref rewrittenReceiver,
+                    ref temps,
+                    ref positionsAssignedToTemp);
+
+                visitedArgumentsBuilder.Add(VisitExpression(arguments[i]));
+
+                foreach (var placeholder in argumentPlaceholders)
+                {
+                    RemovePlaceholderReplacement(placeholder);
+                }
+            }
+
+            Debug.Assert(temps?.Count is null or > 0);
+            return visitedArgumentsBuilder.ToImmutableAndFree();
+
+            void ensureTempTrackingSetup([NotNull] ref ArrayBuilder<LocalSymbol>? temps, ref BitVector positionsAssignedToTemp)
+            {
+                if (temps == null)
+                {
+                    temps = ArrayBuilder<LocalSymbol>.GetInstance();
+                    positionsAssignedToTemp = BitVector.Create(arguments.Length);
+                }
+            }
+
+            ImmutableArray<BoundInterpolatedStringArgumentPlaceholder> setupInterpolationDataIfNeeded(
```

Comment:
nit: `addInterpolationPlaceholderReplacements` to align with `AddPlaceholderReplacement` and `RemovePlaceholderReplacement`?